servers/republik/migrations/sqls/20181119124113-cancellations-up.sql

```diff
@@ -0,0 +1,30 @@
+CREATE DOMAIN cancel_category AS TEXT
+  CHECK(
+    VALUE IN ('EDITORIAL', 'NO_TIME', 'VAUNT', 'LOGIN', 'PAPER', 'NO_AUTO_RENEWAL', 'TOO_EXPENSIVE', 'OTHER')
+  );
+
+CREATE TABLE "membershipCancellations" (
+  "id"              uuid primary key not null default uuid_generate_v4(),
+  "membershipId"    uuid not null references "memberships",
+  "reason"          text,
+  "category"        cancel_category not null,
+  "createdAt"       timestamptz default now(),
+  "updatedAt"       timestamptz default now()
```

Comment:
Architectural: Here we might have a chance to keep track when a cancellation was nulled or revoked. Maybe add `revokedAt` and fill when a cancellation becomes `active` again.

It does keep track of cancellations for `MONTHLY_ABO`, too, si?

Reply:
If I am not mistaken, there would be only one cancellation record per membership. So, if a membership is reactivate, cancellation record is nilled.

My intend however was to have a cancellation reasons "log".

Say membership A. I cancel in March, Seibt did annoy me. I revoke that cancellation in April, after Philipp persuaded me to do so. Later that year, in November, I want to cancel the membership for good.

`membershipCancellations` table would have two records:

```
membershipId | reason            | created | revoked
-------------|-------------------|---------|--------
membership A | "Seibt ist doof"  | 2018-03 | 2018-04
membership A | "ZEIT ONLINE Abo" | 2018-11 | null
```

Since schedule is tight, we can drop this.